assets/js/blocks/cart-checkout/cart/frontend.js

```diff
@@ -2,31 +2,44 @@
  * External dependencies
  */
 import { withRestApiHydration } from '@woocommerce/block-hocs';
+import { useCollection } from '@woocommerce/base-hooks';
+import { RawHTML } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import FullCart from './full-cart';
 import renderFrontend from '../../../utils/render-frontend.js';
 
-const isCartEmpty = false; // @todo check if the cart has some products
-const selector = '.wp-block-woocommerce-cart';
-
-if ( ! isCartEmpty ) {
-	const getProps = () => {
-		return {
-			attributes: {},
-		};
-	};
+/**
+ * Wrapper component to supply API data and show empty cart view as needed.
+ */
+const CartFrontend = ( { emptyCart } ) => {
+	const { results: cartData, isLoading } = useCollection( {
+		namespace: '/wc/store',
+		resourceName: 'cart',
+	} );
 
-	renderFrontend( selector, withRestApiHydration( FullCart ), getProps );
-} else {
-	const containers = document.querySelectorAll( selector );
+	const cartItems = isLoading ? [] : cartData.items;
+	const isCartEmpty = ! isLoading && cartItems.length <= 0;
 
-	if ( containers.length ) {
-		// Use Array.forEach for IE11 compatibility.
-		Array.prototype.forEach.call( containers, ( el ) => {
-			el.classList.remove( 'is-loading' );
-		} );
+	if ( isLoading ) {
```

Comment:
This could go before defining `cartItems` and `isCartEmpty`, no? This way in the definitions you don't need to use a ternary because you will be sure that `isLoading=false`.

Reply:
👍 Would result in good code reduction.